src/ucs/config/parser.c

```diff
@@ -2182,24 +2193,28 @@ void ucs_config_parser_print_opts(FILE *stream, const char *title, const void *o
     if (flags & UCS_CONFIG_PRINT_CONFIG) {
         table_prefix_elem.prefix = table_prefix ? table_prefix : "";
         ucs_list_add_tail(&prefix_list, &table_prefix_elem.list);
-        ucs_config_parser_print_opts_recurs(stream, opts, fields, flags,
-                                            prefix, &prefix_list);
-    }
-
-    if (flags & UCS_CONFIG_PRINT_HEADER) {
-        fprintf(stream, "\n");
+        ucs_config_parser_print_opts_recurs(stream, opts, fields, flags, prefix,
+                                            &prefix_list, title, &printed,
+                                            filter);
     }
 }
 
 void ucs_config_parser_print_all_opts(FILE *stream, const char *prefix,
                                       ucs_config_print_flags_t flags,
-                                      ucs_list_link_t *config_list)
+                                      ucs_list_link_t *config_list,
+                                      const char *filter)
 {
     ucs_config_global_list_entry_t *entry;
     ucs_status_t status;
     char title[64];
     void *opts;
 
+    if (flags & UCS_CONFIG_PRINT_DOC) {
```

Comment:
why moved this block?

Reply:
This was printed multiple times, redundantly, which looked like a bug; now it is only printed once in the beginning.